### PR TITLE
ddl, lightning: use TiDB PD client's discovery to create local backend

### LINKED ISSUE #51103: add index failed with error “Error 1105 (HY000): [Lightning:PD:ErrCreatePDClient]create pd client error: [pd] failed to get cluster id” when inject network partition between ddl owner and pd leader
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、add index
2、inject network partition between ddl owner and pd leader last for 3mins

### 2. What did you expect to see? (Required)
add index can success

### 3. What did you see instead (Required)
add index failed at 2024-02-11 18:36:51 (Error 1105 (HY000): [Lightning:PD:ErrCreatePDClient]create pd client error: [pd] failed to get cluster id)
operatorLogs:
[2024-02-11 18:36:27] ###### start adding index
alter table sbtest1 add index index_test_1707647787882 (c)
[2024-02-11 18:36:27] ###### wait for ddl job finish

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v8.0.0-alpha
Edition: Community
Git Commit Hash: 400bb2c3d748dcb1b111b727fb8bfc072963aba2
Git Branch: heads/refs/tags/v8.0.0-alpha
UTC Build Time: 2024-02-10 11:57:26
GoVersion: go1.21.6
Race Enabled: false
Check Table Before Drop: false
Store: unistore
2024-02-11T18:27:15.287+0800

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #51103

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > chaos test should stop failing

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/importer/table_import.go [hunk 1]
```diff
@@ -220,10 +221,8 @@ func NewTableImporter(param *JobImportParam, e *LoadDataController, id string) (
 	}
 
 	backendConfig := e.getLocalBackendCfg(tidbCfg.Path, dir)
-
-	// todo: use a real region size getter
```
Comment (lance6716) [inline pkg/executor/importer/table_import.go:224]: works well without it, so I delete it.

**Review by tangenta — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

**Comment by lance6716** [comment]
/hold

fixing lightning CI

**Pull request merged**